src/main/scala/edu/ie3/powerFactory2psdm/model/powerfactory/GridModel.scala

```diff
@@ -47,8 +61,12 @@ object GridModel extends LazyLogging {
       logger.debug("There are no switches in the grid.")
       List.empty[Trafos2w]
     })
+    val rawTrafoTpyes2W = rawGrid.trafoTypes2w.getOrElse({
+      logger.debug("There are no 2w trafo types in the grid.")
+      List.empty[TrafoTypes2w]
+    })
 
-    val models = rawNodes ++ rawLines ++ rawLineTypes ++ rawSwitches ++ rawTrafos2W
+    val models = rawNodes ++ rawLines ++ rawLineTypes ++ rawSwitches ++ rawTrafos2W ++ rawTrafoTpyes2W
```

Comment:
On the long run, it would be beneficial, if all those models would have a common trait or abstract class at the top of the hierarchy, because then you could access the id without additional pattern matching.

Reply:
Only way would be to hardcode that into the schema generator which I think is not a good idea as we want that one to work generically - don't we?